### PR TITLE
Fix bug in TensAdd.contract_delta

### PR DESCRIPTION
**Pull request by Kishore96in**
#### Brief description of what is fixed or changed
The current implementation of `TensAdd.contract_delta` assumes that a TensAdd cannot have non-TensExpr arguments. A counter-example is `1 + K(p)*K(q)*delta(-p,-q)`. This PR allows such expressions to be handled without raising an error.

#### Other comments
I have added a function `contract_delta` (sympy.tensor.tensor.contract_delta). The rationale is similar to that for the existing `contract_metric` function.

Suggested reviewers: @Upabjojr 

#### Release Notes

* tensor
  * `TensAdd.contract_delta` no longer raises an error for non-TensExpr arguments
  * A `contract_delta` function has been added that will call `contract_delta` only if the argument is a `TensExpr`

### CODE REVIEW

File: sympy/tensor/tensor.py [hunk 1]
```diff
@@ -2666,7 +2666,7 @@ def __getitem__(self, item):
             return self.data[item]
 
     def contract_delta(self, delta):
-        args = [x.contract_delta(delta) for x in self.args]
+        args = [contract_delta(x, delta) for x in self.args]
```
Comment (smichr) [inline sympy/tensor/tensor.py:2669]:
```suggestion
        args = [x.contract_delta(delta) if isinstance(x, TensExpr) else x for x in self.args]
```

Comment (smichr) [reply sympy/tensor/tensor.py:2669]: With that, another function is not ncecessary

File: sympy/tensor/tensor.py [hunk 2]
```diff
@@ -5235,6 +5235,11 @@ def contract_metric(t, g):
         return t.contract_metric(g)
     return t
 
+def contract_delta(t, delta):
+    if isinstance(t, TensExpr):
+        return t.contract_delta(delta)
+    else:
+        return t
```
Comment (smichr) [inline sympy/tensor/tensor.py:5243]:
```suggestion
```

**Comment by Kishore96in** [comment]
@smichr I have now edited the changelog ( https://github.com/sympy/sympy/wiki/Release-Notes-for-1.14 ) to account for the fact that the `contract_delta` function which was in my original PR has not been added.

**Pull request merged**